File: benchmarks/pandas/bench_ewm_var.py
"""Benchmark: EWM var (span=20) on 100k-element Series"""
import json, time
import numpy as np
import pandas as pd

ROWS = 100_000
WARMUP = 3
ITERATIONS = 10

data = np.sin(np.arange(ROWS) * 0.01)
s = pd.Series(data)

for _ in range(WARMUP):
    s.ewm(span=20).var()

start = time.perf_counter()
for _ in range(ITERATIONS):
    s.ewm(span=20).var()
total = (time.perf_counter() - start) * 1000

print(json.dumps({
    "function": "ewm_var",
    "mean_ms": total / ITERATIONS,
    "iterations": ITERATIONS,
    "total_ms": total,
}))

File: benchmarks/tsb/bench_ewm_var.ts
/**
 * Benchmark: ewm var (alpha=0.1) on 100k-element Series
 */
import { Series } from "../../src/index.js";

const ROWS = 100_000;
const WARMUP = 3;
const ITERATIONS = 10;

const data = Float64Array.from({ length: ROWS }, (_, i) => Math.sin(i * 0.01));
const s = new Series(data);

for (let i = 0; i < WARMUP; i++) {
  s.ewm({ alpha: 0.1 }).var();
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  s.ewm({ alpha: 0.1 }).var();
}
const total = performance.now() - start;

console.log(JSON.stringify({ function: "ewm_var", mean_ms: total / ITERATIONS, iterations: ITERATIONS, total_ms: total }));
